src/jit/emitxarch.cpp

```diff
@@ -2971,12 +2987,20 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = lclVar->AsLclVarCommon()->GetLclNum();
         offset = 0;
     }
+    else if (tmpDsc != nullptr)
+    {
+        varNum = tmpDsc->tdTempNum();
+        offset = 0;
+    }
 
-    if (varNum != BAD_VAR_NUM)
+    // spill temp numbers are negative.
+    if (varNum != BAD_VAR_NUM ||
+        tmpDsc != nullptr)
     {
```

Comment:
After the above, won't it always be the case that `varNum != BAD_VAR_NUM` if `tmpDsc != nullptr`, so the second clause is unnecessary? (Even if it is negative, it should never be BAD_VAR_NUM)